src/storage.ts

```diff
@@ -29,86 +39,137 @@ const accessKeyId = process.env["AWS_ACCESS_KEY_ID"];
-  throw new Error(`Unknown DRIVE_DISK value: '${driveDisk}'`);
+} else if (driveDiskEnv.toLowerCase() === "s3") {
+  driveDisk = "s3";
+} else if (driveDiskEnv.toLowerCase() === "fs") {
+  driveDisk = "fs";
+} else {
+  throw new Error(`Unknown DRIVE_DISK value: '${driveDiskEnv}'`);
 }
-export const DRIVE_DISK: DriveDisk =
-  driveDisk === "fs" || driveDisk === "s3" ? driveDisk : "s3";
 
-const assetUrlBase =
+export const DRIVE_DISK: DriveDisk = driveDisk;
+
+if (
   // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-  process.env["ASSET_URL_BASE"] ??
+  process.env["ASSET_URL_BASE"] !== undefined &&
   // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-  (driveDisk === "s3" ? process.env["S3_URL_BASE"] : undefined);
-if (driveDisk !== "fs" && assetUrlBase == null)
-  throw new Error("ASSET_URL_BASE is required unless DRIVE_DISK=fs.");
-// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-if (driveDisk === "s3" && process.env["ASSET_URL_BASE"] == null) {
-  logger.warn("S3_URL_BASE is deprecated; use ASSET_URL_BASE instead.");
+  process.env["STORAGE_URL_BASE"] === undefined
+) {
+  logger.warn("ASSET_URL_BASE is deprecated; use STORAGE_URL_BASE instead.");
+  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+  process.env["STORAGE_URL_BASE"] = process.env["ASSET_URL_BASE"];
 }
 
-export function getAssetUrl(path: string, base: URL | string): string {
-  if (assetUrlBase == null || assetUrlBase.trim() === "") {
-    return new URL(`/assets/${path.replace(/^\/+/, "")}`, base).href;
-  }
-  return `${assetUrlBase.replace(/\/+$/, "")}/${path.replace(/^\/+/, "")}`;
+if (
+  driveDisk === "s3" &&
+  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+  process.env["S3_URL_BASE"] !== undefined &&
+  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+  process.env["STORAGE_URL_BASE"] === undefined
+) {
+  logger.warn("S3_URL_BASE is deprecated; use STORAGE_URL_BASE instead.");
+  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+  process.env["STORAGE_URL_BASE"] = process.env["S3_URL_BASE"];
 }
 
-let driver: DriverContract;
-switch (DRIVE_DISK) {
-  case "fs":
-    if (!assetPath) {
-      throw new Error("FS_ASSET_PATH is required");
-    }
-    if (!lstatSync(assetPath).isDirectory()) {
-      throw new Error(`Asset path must point to a directory: ${assetPath}`);
-    }
-    access(
-      assetPath,
-      constants.F_OK | constants.R_OK | constants.W_OK,
-      (err) => {
-        if (err) {
-          throw new Error(`${assetPath} must be readable and writable`);
-        }
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+const storageUrlBase = process.env["STORAGE_URL_BASE"];
+
+if (!storageUrlBase) {
+  logger.error("STORAGE_URL_BASE is required");
+  process.exit(1);
```

Comment:
This should probably throw an error instead of logging and then exiting, that would be better caught by the test runner.